utils/bluez-tools/files/bt-agent.init

```diff
@@ -9,29 +11,33 @@ SERVICE_PID_FILE=/var/run/$SERVICE_NAME/pid
 DESC="Bluetooth pairing agent"
 DAEMON=/usr/bin/$SERVICE_NAME
 
-
-# Exit if the package is not installed
-[ -x "$DAEMON" ] || exit 0
-
 AGENT_PIN_FILE=/etc/bt-agent.cfg
-AGENT_OPTS="-c NoInputNoOutput -p $AGENT_PIN_FILE"
-
-#
-# Function that starts the daemon/service
-#
-start()
-{
-	service_start $DAEMON $AGENT_OPTS &
+AGENT_CAPABILITIES=NoInputNoOutput
+
+config_cb() {
+    local type="$1"
+    local name="$2"
+    if [ "$type" = "bt-agent" ]
+    then
+        option_cb() {                                            
+            local option="$1"                                
+            local value="$2"                                 
+            echo "${option//_/-} $value" >> /etc/bt-agent.cfg
+        }
+    else
+        option_cb() { return; }
+    fi
 }
 
-#
-# Function that stops the daemon/service
-#
-stop()
+start_service()
 {
-	service_stop $DAEMON
-	#killall $SERVICE_NAME
-	# Many daemons don't delete their pidfiles when they exit.
-	rm -f $SERVICE_PID_FILE
+	config_load bt-agent
+	
+	procd_open_instance
+	procd_set_param command "$DAEMON"
+	procd_append_param command "-c" "$AGENT_CAPABILITIES"
+	procd_append_param command "-p" "$AGENT_PIN_FILE"
+	procd_append_param command "&"
```

Comment:
You don't have to add `&`. The procd takes care that the program runs as a service.